### PR TITLE
fix: better handling of kwargs inside of contract calls

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did
Use parsing of `kwargs` that avoids converting `AddressType` to `bytes` via the `HexConverter`

fixes: #610

### How I did it
I noticed that the issue raised on #610 had to do with the conversion of `kwargs` inside of `src/ape/contracts/base.py`

### How to verify it
Visual inspection, run any tests necessary

### Checklist

- [x] All changes are completed
- [x] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

**Review by unparalleled-js — COMMENTED** [review]
Is this a refactor and not a fix? I am not sure what bug is being fixed here. It seems more like an optimization?

Also, are we still doing this?

**Comment by fubuloubu** [comment]
> Is this a refactor and not a fix? I am not sure what bug is being fixed here. It seems more like an optimization?

I believe this solves an issue where the converter was turning address strings into bytes

I had thought that was the root cause of another issue I was facing, but it turns out it wasn't the root cause.

File: src/ape/contracts/base.py [hunk 1]
```diff
@@ -17,6 +17,17 @@
     from ape.managers.networks import NetworkManager
 
 
+def _convert_kwargs(kwargs, converter):
```
[the remark this answers is not in the document]
Comment (fubuloubu) [reply src/ape/contracts/base.py:20]: `converter` is not that straightforward of a mypy type

**Comment by unparalleled-js** [comment]
Does this still fix issue #610 ?

**Comment by fubuloubu** [comment]
> Does this still fix issue #610 ?

Yes!

**Review by unparalleled-js — APPROVED** [review]
one, done, and on to the next one!

**Pull request merged**